scripts/get_raw_eumetsat_data.py

```diff
@@ -0,0 +1,354 @@
+
+
+NATIVE_FILESIZE_MB = 102.210123
+CLOUD_FILESIZE_MB = 3.445185
+RSS_ID = "EO:EUM:DAT:MSG:MSG15-RSS"
+CLOUD_ID = "EO:EUM:DAT:MSG:RSS-CLM"
+
+format_dt_str = lambda dt: pd.to_datetime(dt).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+def validate_date(ctx, param, value):
+    try:
+        return format_dt_str(value)
+    except ValueError:
+        raise click.BadParameter("Date must be in format accepted by pd.to_datetime()")
+
+
+@click.command()
+@click.option(
+    "--download_directory",
+    "--dir",
+    default="/run/media/jacob/data/",
+    help="Where to download the data to. Also where the script searches for previously downloaded data.",
+)
+@click.option(
+    "--start_date",
+    "--start",
+    default="2021-01-01",
+    prompt="Starting date to download data, in format accepted by pd.to_datetime()",
+    callback=validate_date,
+)
+@click.option(
+    "--end_date",
+    "--end",
+    default="2021-01-10",
+    prompt="Ending date to download data, in format accepted by pd.to_datetime()",
+    callback=validate_date,
+)
+@click.option(
+    "--backfill",
+    "-b",
+    default=False,
+    prompt="Whether to download any missing data from the start date of the data on disk to the end date",
+    is_flag=True,
+)
+@click.option(
+    "--user_key",
+    "--key",
+    default=None,
+    help="The User Key for EUMETSAT access",
+)
+@click.option(
+    "--user_secret",
+    "--secret",
+    default=None,
+    help="The User secret for EUMETSAT access",
+)
+@click.option(
+    "--auth_filename",
+    default="auth.yaml",
+    help="The auth file containing the user key and access key for EUMETSAT access",
+)
+@click.option(
+    "--bandwidth_limit",
+    "--bw_limit",
+    default=0.0,
+    prompt="Bandwidth limit, in MB/sec, currently ignored",
+    type=float,
+)
+def download_eumetsat_data(
+    download_directory,
+    start_date: str,
+    end_date: str,
+    backfill: bool = False,
+    bandwidth_limit: Optional[float] = None,
+    user_key: Optional[str] = None,
+    user_secret: Optional[str] = None,
+    auth_filename: Optional[str] = None,
+):
+    """
+    Downloads EUMETSAT RSS and Cloud Masks to the given directory,
+     checking first to see if the requested files are already downloaded
+
+    Args:
+        download_directory: Directory to download the files and store them
+        start_date: Start date, in a format accepted by pandas to_datetime()
+        end_date: End date, in a format accepted by pandas to_datetime()
+        backfill: Whether to backfill between the beginning of EUMETSAT data and now, overrides start and end date
+        bandwidth_limit: Bandwidth limit, currently unused
+        user_key: User key for the EUMETSAT API
+        user_secret: User secret for the EUMETSAT API
+        auth_filename: Path to a file containing the user_secret and user_key
+
+    """
+    # Get authentication
+    if auth_filename is not None:
+        user_key, user_secret = load_key_secret(auth_filename)
```

Comment:
Just to be really careful, it might be good throw an error if `auth_filename` is set _and_ (`user_key` or `user_secret` are already set).  Otherwise the user might be confused why `user_key` and/or `user_secret` are being ignored :)

e.g., something like:

```python
if auth_filename is not None:
    if (user_key is not None) or (user_secret is not None):
        raise RuntimeError("Please do not set BOTH auth_filename AND user_key or user_secret!")
    user_key, user_secret = load_key_secret(auth_filename)
```

Reply:
Done!